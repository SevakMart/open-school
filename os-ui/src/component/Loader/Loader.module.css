.mainContent {
  width: 100%;
}
.mainContent .loader {
  width: 7%;
  margin: auto;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}
.mainContent .loader > span {
  background-color: black;
  border-radius: 50%;
  animation: bouncingBall 0.6s infinite alternate;
}
.mainContent .loader > span:nth-child(2) {
  animation-delay: 0.2s;
}
.mainContent .loader > span:nth-child(3) {
  animation-delay: 0.4s;
}
@keyframes bouncingBall {
  from {
    width: 10px;
    height: 10px;
    opacity: 1;
    transform: translate3d(0);
  }
  to {
    width: 20px;
    height: 20px;
    opacity: 0.3;
    transform: translate3d(0, -20px, 0);
  }
}/*# sourceMappingURL=Loader.module.css.map */