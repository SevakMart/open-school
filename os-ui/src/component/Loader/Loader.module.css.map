{"version":3,"sources":["Loader.module.scss","Loader.module.css","../../styles/Mixins/FlexMixins.scss"],"names":[],"mappings":"AAGA;EACI,WAAA;ACFJ;ADIA;EACI,SAAA;EACA,YAAA;EEPF,aAAA;EACA,mBFOqB;EENrB,mBFMyB;EELzB,8BFKgC;ACClC;ADEA;EACI,uBAAA;EACA,kBAAA;EACA,+CAAA;ACAJ;ADEA;EACI,qBAAA;ACAJ;ADEA;EACI,qBAAA;ACAJ;ADEA;EACI;IACI,WAAA;IACA,YAAA;IACA,UAAA;IACA,yBAAA;ECAN;EDEE;IACI,WAAA;IACA,YAAA;IACA,YAAA;IACA,mCAAA;ECAN;AACF","file":"Loader.module.css"}