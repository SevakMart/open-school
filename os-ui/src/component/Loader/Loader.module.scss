@import "../../styles/Mixins/FlexMixins.scss";


.mainContent{
    width:100%;

.loader {
    width:7%;
    margin:auto;
    @include flexStyle(row,center,space-between);
}

.loader >span {
    background-color:black;
    border-radius:50%;
    animation:bouncingBall 0.6s infinite alternate;
}
.loader>span:nth-child(2){
    animation-delay:0.2s;
}
.loader>span:nth-child(3){
    animation-delay:0.4s;
}
@keyframes bouncingBall {
    from {
        width:10px;
        height:10px;
        opacity:1;
        transform:translate3d(0)
    }
    to{
        width:20px;
        height:20px;
        opacity:0.3;
        transform:translate3d(0,-20px,0);
    }
}
}