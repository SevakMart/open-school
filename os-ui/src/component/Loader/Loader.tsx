import styles from './Loader.module.scss';

const Loader = () => {
  const { mainContent, loader } = styles;
  return (
    <div className={mainContent}>
      <div className={loader}>
        <span />
        <span />
        <span />
      </div>
    </div>
  );
};
export default Loader;
