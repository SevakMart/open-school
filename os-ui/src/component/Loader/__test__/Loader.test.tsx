import { render } from '@testing-library/react';
import Loader from '../Loader';

test('Create a Snapshot testing', () => {
  const { asFragment } = render(<Loader />);
  expect(asFragment()).toMatchSnapshot();
});
