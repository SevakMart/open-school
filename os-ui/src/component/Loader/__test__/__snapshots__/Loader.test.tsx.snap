// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create a Snapshot testing 1`] = `
<DocumentFragment>
  <div
    class="mainContent"
  >
    <div
      class="loader"
    >
      <span />
      <span />
      <span />
    </div>
  </div>
</DocumentFragment>
`;
